### PR TITLE
fix(tabs): new tab animation broken by code to support IE11

### LINKED ISSUE #11689: tabs: wrong tab position when a new tab is added
## Bug, enhancement request, or proposal:

There is an error with tabs position when you add a new tab and it does not fit and needs to be horizontally scrolled.

# CodePen and steps to reproduce the issue:
#### Detailed Reproduction Steps: 
1. Go to https://material.angularjs.org/HEAD/demo/tabs
2. Go to "Dyanmic tabs" block.
3. Add a new tab with any label and any content.

### What is the expected behavior?
A new tab is created and tabs are scrolled to it. Then you can go back with the left arrow until first tab.

### What is the current behavior?
A new tab is created and tabs are scrolled to it. You cannot see any tab until you click on left arrow, then you see some tabs and you cannot go to previous tabs.

### What is the use-case or motivation for changing an existing behavior?
I want to create new tabs dynamically and I want to be able to go to first tab.

### Which versions of AngularJS, Material, OS, and browsers are affected?

- AngularJS: 1.7.5
- AngularJS Material: 1.1.14
- OS: Ubuntu 18.04
- Browsers: Chome v72.0.3626.119
 
### Is there anything else we should know? Stack Traces, Screenshots, etc.
This bug occurs since AngularJS Material v1.1.11, with v1.1.10 it is working correctly.

### PR DESCRIPTION
**Pull request by Splaktar**
## PR Checklist
Please check that your PR fulfills the following requirements:
- [x] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [ ] Tests for the changes have been added or this is not a bug fix / enhancement
- [x] Docs have been added, updated, or were not required

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Enhancement
[ ] Documentation content changes
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
The scroll to new tab animation is broken.

Issue Number: 
Fixes #11689

## What is the new behavior?
The scroll to new tab animation is fixed for browsers other than IE.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/core/util/util.js [hunk 1]
```diff
@@ -901,7 +901,14 @@ function UtilFactory($document, $timeout, $compile, $rootScope, $$mdAnimate, $in
...
       return serializer.serializeToString(element);
-    }
+    },
+
+    /**
+     * Support: IE 9-11 only
+     * documentMode is an IE-only property
+     * http://msdn.microsoft.com/en-us/library/ie/cc196988(v=vs.85).aspx
+     */
+    msie: window.document.documentMode
```
Comment (Splaktar) [inline src/core/util/util.js]: Taken from [AngularJS](https://github.com/angular/angular.js/blob/master/src/Angular.js#L165-L170)

File: src/components/tabs/js/tabsController.js [hunk 2]
```diff
@@ -576,17 +579,20 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
...
-    // Work around width calculation issues on IE11 when pagination is enabled
-    if (shouldPaginate) {
-      getElements().paging.style.width = '999999px';
-    } else {
-      getElements().paging.style.width = undefined;
+    // Work around width calculation issues on IE11 when pagination is enabled.
+    // Don't do this on other browsers because it breaks scroll to new tab animation.
+    if ($mdUtil.msie) {
+      if (shouldPaginate) {
+        getElements().paging.style.width = '999999px';
```
Comment (Splaktar) [inline src/components/tabs/js/tabsController.js]: Wrapping this in a check for IE is the actual fix for non-IE browsers.

**Comment by gauravlodha** [comment]
@Splaktar: after this commit, tabs getting added twice the time in mobile devices works fine for desktop

**Comment by Splaktar** [comment]
@gauravlodha I opened https://github.com/angular/material/issues/11725 to track that issue. Thank you for reporting.

**Comment by Splaktar** [comment]
The check is working on Microsoft Edge 44.17763.1.0 and the `md-pagination-wrapper`'s `width` isn't being tweaked, but the new tab animation is still broken in the same way that IE11 is.

**Pull request merged**